substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JavaLangSubstitutions.java

```diff
@@ -722,6 +745,348 @@ static Package getPackage(Class<?> c) {
     }
 }
 
+@TargetClass(className = "java.lang.StackWalker", onlyWith = JDK9OrLater.class)
+@Substitute
+final class Target_java_lang_StackWalker {
+
+    @Alias
+    private final Set<Option> options;
+
+    @TargetClass(className = "java.lang.StackWalker$StackFrame", onlyWith = JDK9OrLater.class)
+    @Substitute
+    interface StackFrame {
+
+        String getClassName();
+
+        String getMethodName();
+
+        Class<?> getDeclaringClass();
+
+        default MethodType getMethodType() {
+            throw new UnsupportedOperationException();
+        }
+
+        default String getDescriptor() {
+            throw new UnsupportedOperationException();
+        }
+
+        int getByteCodeIndex();
+
+        String getFileName();
+
+        int getLineNumber();
+
+        boolean isNativeMethod();
+
+        StackTraceElement toStackTraceElement();
+    }
+
+    @TargetClass(className = "java.lang.StackWalker$Option", onlyWith = JDK9OrLater.class)
+    enum Option {
```

Comment:
Use a regular class `final class Target_java_lang_StackWalker_Option` here. The `@Alias` fields are really just an alias to the `static` fields defined in the original class. You cannot use the `enum` methods or the fact that the class is an enum anyway without bad things happening.

Reply:
OK.